console/console-init/ui/src/modules/address-detail/components/AddressLinksToolbar/AddressLinksToolbar.tsx

```diff
@@ -1,56 +1,36 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
 import React from "react";
 import {
   DataToolbarItem,
   DataToolbar,
   DataToolbarContent,
-  SelectOptionObject,
-  DataToolbarChipGroup,
-  DataToolbarChip,
   DataToolbarContentProps
 } from "@patternfly/react-core";
 import { ISortBy } from "@patternfly/react-table";
 import { SortForMobileView, useWindowDimensions } from "components";
-import { AddressLinksToggleGroup } from "modules/address-detail/components";
+import {
+  AddressLinksToggleGroup,
+  IAddressLinksToggleGroupProps
+} from "modules/address-detail/components";
 
-export interface IAddressLinksToolbarProps extends DataToolbarContentProps {
-  totalRecords: number;
-  filterSelected?: string;
-  nameSelected?: string;
-  nameInput?: string;
-  containerSelected?: string;
-  containerInput?: string;
-  nameOptions?: any[];
-  containerOptions?: any[];
-  roleIsExpanded: boolean;
-  roleSelected?: string | null;
-  selectedNames: Array<{ value: string; isExact: boolean }>;
-  selectedContainers: Array<{ value: string; isExact: boolean }>;
-  onFilterSelect: (value: string) => void;
-  onNameSelect: (e: any, selection: SelectOptionObject) => void;
-  onNameClear: () => void;
-  onNameFilter: (e: any) => any[];
-  onContainerSelect: (e: any, selection: SelectOptionObject) => void;
-  onContainerClear: () => void;
-  onContainerFilter: (e: any) => any[];
-  onRoleToggle: () => void;
-  onRoleSelect: (e: any, selection: SelectOptionObject) => void;
-  onSearch: () => void;
-  onDelete: (
-    category: string | DataToolbarChipGroup,
-    chip: string | DataToolbarChip
-  ) => void;
+export interface IAddressLinksToolbarProps
+  extends IAddressLinksToggleGroupProps {
   sortValue?: ISortBy;
   setSortValue: (value: ISortBy) => void;
   onClearAllFilters: () => void;
 }
-const AddressLinksToolbar: React.FunctionComponent<IAddressLinksToolbarProps> = ({
+const AddressLinksToolbar: React.FunctionComponent<IAddressLinksToolbarProps &
```

Comment:
why there are two types? I thinks we already extended the second interface DataToolbarContentProps in the IAddressLinksToolbarProps.

Reply:
second one interface is the part of PatternFly. as we are using PF component in this file.In previous code this interface extends with component interface but sometimes 2 interfaces cause the issue due to incompatible types. might be we can pass some properties as same name as available in PF interface. that's why we have both interface.